psydac/core/bsplines.py

```diff
@@ -656,23 +665,35 @@ def elevate_knots(knots, degree, periodic):
     periodic : bool
         True if domain is periodic, False otherwise.
 
+    multiplicity : int
+        multiplicity of the knots.
+
     Returns
     -------
     new_knots : 1D numpy.ndarray
         Knots sequence of spline space of degree p+1.
 
     """
 
+    knots = np.array(knots)
+
     if periodic:
         [T, p] = knots, degree
         period = T[-1-p] - T[p]
-        left   = T[-1-p-(p+1)] - period
-        right  = T[   p+(p+1)] + period
+        left   = [T[-1-p-(p+1)] - period]
+        right  = [T[   p+(p+1)] + period]
     else:
-        left  = knots[0]
-        right = knots[-1]
+        left  = [knots[0],*knots[:degree+1]]
+        right = [knots[-1],*knots[-degree-1:]]
+
+        diff   = np.append(True, np.diff(knots[degree+1:-degree-1])>1e-15)
+        if len(knots[degree+1:-degree-1])>0:
+            unique = knots[degree+1:-degree-1][diff]
+            knots  = np.repeat(unique, multiplicity)
+        else:
+            knots = knots[degree+1:-degree-1]
```

Comment:
See my previous comment to the `breakpoints` function: don't hide tolerances inside this function

Reply:
done